lib/usb/usb_f107.c

```diff
@@ -60,8 +60,7 @@ static usbd_device *stm32f107_usbd_init(void)
 	OTG_FS_GRSTCTL |= OTG_GRSTCTL_CSRST;
 	while (OTG_FS_GRSTCTL & OTG_GRSTCTL_CSRST);
 
-	uint32_t id = OTG_FS_CID;
-	if (id == OTG_CID_HAS_VBDEN) {
+	if (OTG_FS_CID == OTG_CID_HAS_VBDEN) {
```

Comment:
Perhaps you are looking at a different set of docs?  My understanding is that Bit 21 polarity has changed.

## STM32F427
DocID018909 Rev 8 1273/1718
RM0090 USB on-the-go full-speed (OTG_FS)
1362
![image](https://cloud.githubusercontent.com/assets/1945821/17515988/7a8609d8-5dd7-11e6-8946-cccc093c710a.png)

## STM32F469

DocID026744 Rev 2 1311/1599 
![image](https://cloud.githubusercontent.com/assets/1945821/17515961/4fec5254-5dd7-11e6-9d7c-087c86b83247.png)

Good point about SDIS - but I prefer the deliberate setting with the comment because it highlights the difference